src/database/schema/UserSchema.ts

```diff
@@ -221,19 +261,44 @@ class UserSchema implements BaseDbSchema {
   }
 
   static fromMemberWebsocketObject(json: any, channelId: string): UserSchema {
-    return new UserSchema({
-      id: uuidv4(),
-      channelId,
-      userId: json?.user?.id,
-      username: json?.user?.name,
-      countryCode: '',
-      createdAt: json?.user?.created_at,
-      updatedAt: json?.updated_at,
-      lastActiveAt: json?.user?.last_active,
-      profilePicture: json?.user?.image,
-      bio: '',
-      isBanned: json?.banned
-    });
+    try {
+      return new UserSchema({
+        id: uuidv4(),
+        channelId,
+        userId: json?.user?.id,
+        username: json?.user?.name,
+        countryCode: '',
+        createdAt: json?.user?.created_at,
+        updatedAt: json?.updated_at,
+        lastActiveAt: json?.user?.last_active,
+        profilePicture: json?.user?.image,
+        bio: '',
+        isBanned: json?.banned,
+        anon_user_info_color_code: json?.anon_user_info_color_code,
+        anon_user_info_color_name: json?.anon_user_info_color_name,
+        anon_user_info_emoji_code: json?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: json?.anon_user_info_emoji_name
+      });
+    } catch (e) {
+      console.log('error on fromMemberWebsocketObject', e);
+      return new UserSchema({
+        id: uuidv4(),
+        channelId,
+        userId: null,
+        username: null,
+        countryCode: null,
+        createdAt: null,
+        updatedAt: null,
+        lastActiveAt: null,
+        profilePicture: null,
+        bio: null,
+        isBanned: null,
+        anon_user_info_color_code: null,
+        anon_user_info_color_name: null,
+        anon_user_info_emoji_code: null,
+        anon_user_info_emoji_name: null
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `fromMemberWebsocketObject` function, you are catching an error but only logging it to the console and returning a new UserSchema object with null values. This might lead to issues later on when you try to use these objects. Consider throwing the error after logging it so that the caller can handle it appropriately.
```diff
  try {
    return new UserSchema({
      id: uuidv4(),
      channelId,
      userId: json?.user?.id,
      username: json?.user?.name,
      countryCode: '',
      createdAt: json?.user?.created_at,
      updatedAt: json?.updated_at,
      lastActiveAt: json?.user?.last_active,
      profilePicture: json?.user?.image,
      bio: '',
      isBanned: json?.banned,
      anon_user_info_color_code: json?.anon_user_info_color_code,
      anon_user_info_color_name: json?.anon_user_info_color_name,
      anon_user_info_emoji_code: json?.anon_user_info_emoji_code,
      anon_user_info_emoji_name: json?.anon_user_info_emoji_name
    });
  } catch (e) {
    console.log('error on fromMemberWebsocketObject', e);
-   return new UserSchema({
-     id: uuidv4(),
-     channelId,
-     userId: null,
-     username: null,
-     countryCode: null,
-     createdAt: null,
-     updatedAt: null,
-     lastActiveAt: null,
-     profilePicture: null,
-     bio: null,
-     isBanned: null,
-     anon_user_info_color_code: null,
-     anon_user_info_color_name: null,
-     anon_user_info_emoji_code: null,
-     anon_user_info_emoji_name: null
-   });
+   throw e;
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->